cranelift/codegen/src/isa/aarch64/inst/mod.rs

```diff
@@ -290,6 +290,10 @@ pub enum VecALUOp {
     Umlal,
     /// Zip vectors (primary) [meaning, high halves]
     Zip1,
+    /// Signed multiply long (low halves)
```

Comment:
Yeah, that sounds like a nice cleanup for a followup.  I'm curious though .. when you say "handling of the high and the low halves can be streamlined", what did you have in mind, roughly?

Reply:
Have a look at `Inst::VecMiscNarrow`, which organizes things similarly, but for narrowing operations. TBH I don't have an exact code change in mind (as I said, neither Joey, nor I did anything yet because there was no real need), but having a separate boolean parameter to specify the half is probably going to be the way to do it.

I dislike the ISA's approach (via the vector shape), but it has constraints (encoding space) that are not applicable to the `Inst` enum. Anyway, the `Inst` variants don't necessarily map 1:1 to machine instructions, so we can use that to improve ergonomics (e.g. we support bitwise operations on arbitrary vector shapes, not just 8-bit elements).